lib/reversetunnel/remotesite.go

```diff
@@ -755,15 +755,21 @@ func (s *remoteSite) DialAuthServer(params reversetunnelclient.DialParams) (net.
 // located in a remote connected site, the connection goes through the
 // reverse proxy tunnel.
 func (s *remoteSite) Dial(params reversetunnelclient.DialParams) (net.Conn, error) {
-	recConfig, err := s.localAccessPoint.GetSessionRecordingConfig(s.ctx)
+	localRecCfg, err := s.localAccessPoint.GetSessionRecordingConfig(s.ctx)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	remoteRecCfg, err := s.remoteAccessPoint.GetSessionRecordingConfig(s.ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	remoteProxyRec := services.IsRecordAtProxy(remoteRecCfg.GetMode())
 
-	// If the proxy is in recording mode and a SSH connection is being
-	// requested or the target server is a registered OpenSSH node, build
-	// an in-memory forwarding server.
-	if shouldDialAndForward(params, recConfig) {
+	// If the remote cluster is recording at the proxy we need to respect
+	// that and forward and record the session. We will be connecting
+	// to the node without connecting through the remote Proxy, so the
+	// session won't have a chance to get recorded at the remote Proxy.
+	if remoteProxyRec || shouldDialAndForward(params, localRecCfg) {
```

Comment:
Maybe it can be beneficial to split this condition and call `return s.dialAndForward(params)` separately for each part? That way we don't need to call `s.remoteAccessPoint.GetSessionRecordingConfig(s.ctx)` if local one satisfies the condition.

Reply:
I don't understand this comment. Shouldn't the remote proxy be responsible for the last leg of the connection and thus for the recording?

Reply:
@espadolini as I understood it when connecting to a leaf node through a root proxy the root cluster should have the recording. Here are the items in the test plan that lead me to finding this issue: 

- [Sessions can be recorded at the proxy](https://goteleport.com/docs/server-access/guides/recording-proxy-mode/)
  - Sessions on remote clusters are recorded in the local cluster